geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java

```diff
@@ -68,46 +77,40 @@ public Object get(
 		return properties.get(toPropertyName(propertyName));
 	}
 
-	public void store(
+	public synchronized void store(
 			final ParameterEnum propertyName,
 			final Object value ) {
-		synchronized (MUTEX) {
-			properties.put(
-					toPropertyName(propertyName),
-					value);
-		}
+		properties.put(
+				toPropertyName(propertyName),
+				value);
 	}
 
-	public Object storeIfEmpty(
+	public synchronized Object storeIfEmpty(
 			final ParameterEnum propertyName,
 			final Object value ) {
 		final String pName = toPropertyName(propertyName);
-		synchronized (MUTEX) {
-			if (!properties.containsKey(pName)) {
-				LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
-				properties.put(
-						pName,
-						value);
-				return value;
-			}
-
-			return properties.get(pName);
+		if (!properties.containsKey(pName)) {
+			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
```

Comment:
Instead of string concatenation, move the LOGGER statements over to use dynamic content injection {}
ref: http://slf4j.org/faq.html#logging_performance
String won't even be built if the log level is high enough.